app/services/dunning_campaigns/update_service.rb

```diff
@@ -45,21 +32,84 @@ def call
 
     attr_reader :dunning_campaign, :organization, :params
 
+    def update_dunning_campaign_attributes
+      dunning_campaign.assign_attributes(permitted_attributes)
+    end
+
+    def permitted_attributes
+      params.slice(:name, :code, :description, :days_between_attempts, :max_attempts)
+    end
+
     def handle_thresholds
+      thresholds_updated = false
+
       input_threshold_ids = params[:thresholds].map { |t| t[:id] }.compact
 
       # Delete thresholds not included in the payload
-      dunning_campaign.thresholds.where.not(id: input_threshold_ids).discard_all
+      discarded_thresholds = dunning_campaign
+        .thresholds
+        .where.not(id: input_threshold_ids)
+        .discard_all
+
+      thresholds_updated = discarded_thresholds.present?
 
       # Update or create new thresholds from the input
       params[:thresholds].each do |threshold_input|
-        dunning_campaign.thresholds.find_or_initialize_by(
+        threshold = dunning_campaign.thresholds.find_or_initialize_by(
           id: threshold_input[:id]
-        ).update!(
-          amount_cents: threshold_input[:amount_cents],
-          currency: threshold_input[:currency]
         )
+
+        threshold.assign_attributes(threshold_input.slice(:amount_cents, :currency))
+
+        thresholds_updated ||= threshold.changed? && threshold.persisted?
+        threshold.save!
       end
+
+      reset_customers_if_no_threshold_match if thresholds_updated
+    end
+
+    def reset_customers_if_no_threshold_match
+      customers_applied_campaign = organization
+        .customers
+        .with_dunning_campaign_not_completed
+        .where(applied_dunning_campaign: dunning_campaign)
+
+      customers_fallback_campaign = organization
+        .customers
+        .falling_back_to_default_dunning_campaign
+        .with_dunning_campaign_not_completed
+        .where(dunning_campaign.applied_to_organization ? nil : "1 = 0")
```

Comment:
not sure to understand 🤔

Reply:
it can be removed, this was a left over from initial query implementation, thanks for raising it.